pythran/pythonic/itertools/ifilter.hpp

```diff
@@ -95,7 +95,8 @@ namespace pythonic
       template <typename Operator, typename List0>
       ifilter<Operator, List0>::ifilter(Operator _op, List0 const &_seq)
           : utils::iterator_reminder<false, List0>(_seq),
-            iterator(_op, this->value), end_iter(npos(), _op, this->value)
+            iterator(_op, utils::iterator_reminder<false, List0>::value),
```

Comment:
instead of true/false you could use two different empty structures with explicit names, that would make the review easier